src/procfs.cpp

```diff
@@ -137,6 +140,50 @@ uptime procfs::get_uptime() const
     return parsers::parse_uptime_line(line);
 }
 
+sys_stat procfs::get_stat() const
+{
+    static const std::string STAT_FILE("stat");
+    auto path = _root + STAT_FILE;
+
+    std::vector<std::string> lines;
+    parsers::parse_lines(
+        path,
+        std::back_inserter(lines),
+        [&](const std::string &line) { return line; }
+    );
+
+    sys_stat stats;
+
+    for (const std::string& line : lines) {
+        if (line.rfind("cpu", 0) == 0) {
+            stats.cpus.emplace_back(parsers::parse_stat_cpu_line(line));
```

Comment:
Following your suggestion and my comment about the parser, I'm not sure this is a correct assumption. I mean, not sure CPUs are going to be always ordered correctly.
Especially with hot-swapping and such, I wonder how this is going to be handled by the machine...
Maybe we should use the `X` from `cpuX` to add the values to the right index in the vector...